### DISCUSSION TITLE
Verticals and Programs should be included in the content information

### DISCUSSION THREAD

**Discussion by amitpriyadarshi** [Sunbird-Ed/Community#289] [Bugs]
@sharathkashyap @rajeevsathish 
There are few channel specific property against content (eg: Verticals and Programs), which need to be available at the time of reviewing the content. As we discussed the metadata coming in review page are hardcoded. Please add these property as well.

**Comment by Krishnaj20** [comment]
@rajeevsathish - for your attention.

**Reply by rajeevsathish** [reply]
@luckynvdu @vaibhavbhuva @pallakartheekreddy please have a look into this discussion and please let me know if any thing has to be done from SunbirdED BB
cc: @Krishnaj20 @vinukumar-vs

**Comment by vinukumar-vs** [comment]
@amitpriyadarshi Can you please share more details of which metadata page are you taking about while reviewing?

Is it specific to primary category are you talking or for all? Can you please share more details with UI screen attached.

@vaibhavbhuva @gouravmore

**Reply by vinukumar-vs** [reply]
@gouravmore the below defect is handling while save the content metadata. Can you confirm these changes will work for review form or not? 
https://project-sunbird.atlassian.net/browse/KN-238

@vaibhavbhuva

**Reply by gouravmore** [reply]
@vinukumar-vs Yes, This mentioned defect changes will work for review form also.

**Reply by Krishnaj20** [reply]
@gouravmore - In which release will this bug be fixed by Knowlg team?

**Reply by gouravmore** [reply]
@Krishnaj20 This bug has been fixed in release-5.1.0 by knowlg team.

**Reply by Krishnaj20** [reply]
@amitpriyadarshi - If your query is answered pls mark it as answered.

**Reply by kameshbhr** [reply]
@vinukumar-vs @Krishnaj20  The above referenced defect (KN-238) is about the values not showing up in the Editor (for creator).
But what @amitpriyadarshi mentioned in this thread is about Review page for the reviewer.  
New fields can be configured in Editor, but Review page doesn't seem to have this capability. It only shows a hardcoded set of fields. 
Here is the content review page for your reference where the content information is shown:
![image](https://user-images.githubusercontent.com/17846270/207280039-9c9e8605-bb6b-40c0-ae15-fc52181ae097.png)

We need to add Verticals and Programs fields to be shown here. Hope this clarifies.

**Reply by Krishnaj20** [reply]
thanks @kameshbhr for the clarification. 

@luckynvdu @pallakartheekreddy - raised https://github.com/Sunbird-coKreat/Community/discussions/40 for your team to look into it and address it.

**Reply by vinukumar-vs** [reply]
@rajeevsathish presently the content read request itself doesn't have the verticals & programs in the queryParams. 
Please find the below screen.


It should read these properties from the content metadata & render here.  
<img width="1791" alt="Screenshot 2022-12-14 at 2 29 08 PM" src="https://user-images.githubusercontent.com/4496096/207551764-8e5f89c0-def2-4895-b6ea-cb9c12ee4b60.png">

I can see a challenge how this properties will be consistent to the editors is still a challenge. Otherwise any change in the editor metadata form may not reflect in the content metadata while reviewing.

@Krishnaj20 @kameshbhr @vaibhavbhuva

**Reply by vinukumar-vs** [reply]
Again the same metadata properties may be present on consumption side as well(mobile/portal).

**Reply by kameshbhr** [reply]
I think the right solution is to have a single configuration  for the set of metadata fields to be shown in the UI for the entire creation flow - edit/preview/review UI.

**Comment by kameshbhr** [comment]
@Krishnaj20 @vinukumar-vs Any update on this issue? Is it planned for any release?

**Reply by Krishnaj20** [reply]
@kameshbhr - We will tag this to the next release (5.3.0).
